services/extract-stats.js

```diff
@@ -1,20 +1,42 @@
 import * as cheerio from "cheerio";
 
 function extractStats(html) {
-	const $ = cheerio.load(html);
-
-	const detailsDiv = $("div.d-flex.flex__fl-equal.fw-wrap.gs24").first();
-	const badgeDivs = detailsDiv.find("div.fs-title");
-
-	return {
-		username: $(".flex--item.mb12.fs-headline2.lh-xs").text().trim(),
-		reputation: $("div.fs-body3.fc-black-600").eq(0).text(),
-		peopleReached: $("div.fs-body3.fc-black-600").eq(1).text(),
-		answers: $("div.fs-body3.fc-black-600").eq(2).text(),
-		goldBadges: badgeDivs.eq(0).text().trim(),
-		silverBadges: badgeDivs.eq(1).text().trim(),
-		bronzeBadges: badgeDivs.eq(2).text().trim(),
-	};
+  const $ = cheerio.load(html);
+
+  const detailsDiv = $("div.d-flex.flex__fl-equal.fw-wrap.gs24").first();
+  const badgeDivs = detailsDiv.find("div.fs-title"); 1 
+
+  // Extract badge types and counts using class names (if available)
+  const goldBadges = detailsDiv.find("div.badge-gold").text().trim();
+  const silverBadges = detailsDiv.find("div.badge-silver").text().trim();
+  const bronzeBadges = detailsDiv.find("div.badge-bronze").text().trim();
+
+  // If class names are not found or not reliable, use text content as a fallback
+  if (!goldBadges || !silverBadges || !bronzeBadges) {
```

Comment:
If any one is not available why are we processing for all ? any reason.
Let's say goldbadges is not there silverBadges and bronzeBadges exists.
Then we are recalculating for silverbadges and bronzebadges even. can we try to optimize this , or any particular reason this is written this way?